### PR TITLE
[POAE7-2457] Support Case/IF Function for Arrow Format with String Function

### PR DESCRIPTION
**Pull request by haiweizh**
### What changes were proposed in this pull request?

Support Case/IF Function for Arrow Format with String Function

### Why are the changes needed?

Support Case/IF Function for Arrow Format with String Function

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Uts

### Which label does this PR belong to?

CODEREFACTOR,cider,test

### CODE REVIEW

File: cider/exec/template/CaseIR.cpp [hunk 1]
```diff
@@ -158,11 +158,14 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenCaseExpr(
   std::unique_ptr<CodegenColValues> case_val =
       codegenCaseExpr(case_expr, case_llvm_type, is_real_str, co);
   if (is_real_str) {
-    // FIXME(haiwei): [POAE7-2457] ,blocking by [POAE7-2415]
-    CIDER_THROW(CiderCompileException,
-                "String type case when is not currently supported.");
+    auto case_expr = dynamic_cast<FixedSizeColValues*>(case_val.get());
```
Comment (kd97) [inline cider/exec/template/CaseIR.cpp:161]: should use `TwoValueColValues` type?

Comment (kd97) [reply cider/exec/template/CaseIR.cpp:161]: and may need use new string type encoder and decoder.

Comment (spevenhe) [reply cider/exec/template/CaseIR.cpp:161]: Could just use codegenVarCharColVar?

Comment (haiweizh) [reply cider/exec/template/CaseIR.cpp]: Done.

**Review by yma11 — APPROVED** [review]

**Review by spevenhe — APPROVED** [review]

**Pull request merged**